packages/oauth/src/providers/kakao.ts

```diff
@@ -0,0 +1,169 @@
+import {
+	createOAuth2AuthorizationUrl,
+	OAuth2ProviderAuth,
+	validateOAuth2AuthorizationCode
+} from "../core/oauth2.js";
+import { ProviderUserAuth } from "../core/provider.js";
+import { authorizationHeader, handleRequest } from "../utils/request.js";
+
+import type { Auth } from "lucia";
+
+type Config = {
+	clientId: string;
+	clientSecret: string; // Kakao doesn't require clientSecret but it's recommended to use it
+	redirectUri: string;
+	scope?: string[];
+};
+
+const PROVIDER_ID = "kakao";
+
+export const kakao = <_Auth extends Auth = Auth>(
+	auth: _Auth,
+	config: Config
+): KakaoAuth<_Auth> => {
+	return new KakaoAuth(auth, config);
+};
+
+export class KakaoAuth<_Auth extends Auth = Auth> extends OAuth2ProviderAuth<
+	KakaoUserAuth<_Auth>
+> {
+	private config: Config;
+
+	constructor(auth: _Auth, config: Config) {
+		super(auth);
+
+		this.config = config;
+	}
+
+	public getAuthorizationUrl = async (): Promise<
+		readonly [url: URL, state: string]
+	> => {
+		return await createOAuth2AuthorizationUrl(
+			"https://kauth.kakao.com/oauth/authorize",
+			{
+				clientId: this.config.clientId,
+				scope: this.config.scope ?? [],
+				redirectUri: this.config.redirectUri
+			}
+		);
+	};
+
+	public validateCallback = async (
+		code: string
+	): Promise<KakaoUserAuth<_Auth>> => {
+		const kakaoTokens = await this.validateAuthorizationCode(code);
+		const kakaoUser = await getKakaoUser(kakaoTokens.access_token);
+		return new KakaoUserAuth(this.auth, kakaoUser, kakaoTokens);
+	};
+
+	private validateAuthorizationCode = async (
+		code: string
+	): Promise<KakaoTokens> => {
```

Comment:
The return type (`KakaoTokens`) should have `camelCase` properties (see other providers)

Reply:
Change types to camelCase thanks for the review

Reply:
`validateOAuth2AuthorizationCode()` doesn't transform snake_case into camelCase, so it needs to be done manually